src/c/edgex-rest.c

```diff
@@ -729,6 +741,14 @@ edgex_device *edgex_device_dup (const edgex_device *e)
 {
   edgex_device *result = malloc (sizeof (edgex_device));
   result->name = strdup (e->name);
+  if (e->parent)
```

Comment:
Using ternary operator here could make the code more concise
`result->parent = e->parent ? strdup(e->parent) : NULL;`